src/sentry/incidents/logic.py

```diff
@@ -698,6 +698,16 @@ def nullify_id(model: Model) -> None:
         snuba_query_snapshot: SnubaQuery = deepcopy(_unpack_snuba_query(alert_rule))
         nullify_id(snuba_query_snapshot)
         snuba_query_snapshot.save()
+
+        event_types = SnubaQueryEventType.objects.filter(
+            snuba_query=_unpack_snuba_query(alert_rule)
+        )
+        for event_type in event_types:
+            event_type_snapshot = deepcopy(event_type)
+            nullify_id(event_type_snapshot)
+            event_type_snapshot.snuba_query = snuba_query_snapshot
+            event_type_snapshot.save()
```

Comment:
since this is saving in the loop, will it create an N+1 query? 🤔 i wonder if we could make this a batch operation instead.

Reply:
Most snuba queries likely just have one event type associated with them (with a max of 2 when error and default types are selected) but good callout, I made it a bulk create!